lib/plugins.py

```diff
@@ -545,6 +572,15 @@ def __init__(self, parent, config, name):
                 hooks[aname].append((self, func))
                 self._hooks_i_registered.append((aname,func))
 
+        # collect names of all class attributes with ._is_daemon_command
+        self._daemon_commands = tuple(attrname for attrname in dir(type(self))
+                                      if getattr(getattr(type(self),attrname), '_is_daemon_command',False))
+        # we don't allow conflicting definitions of daemon command (between different plugins)
+        collisions = set(self._daemon_commands).intersection(self.parent.daemon_commands)
+        if collisions:
+            raise RuntimeError('colliding daemon command names in plugin', collisions)
+        self.parent.daemon_commands.update({cmdname:getattr(self,cmdname) for cmdname in self._daemon_commands})
```

Comment:
Hmm. I'd suggest writing to the `self._daemon_commands` dict once you have "committed" the commands to the parent dict and not before (before you can use a temp dict) --  This way if you raised `RuntimeError` above, your instance `dict` will be empty (which is really what you want). Thus, the invariant "if a command is in my local dict, it is also in the global one" is preserved.

Reply:
Additionally: If  `RuntimeError` was raised above and then this plugin is closed -- it will *clobber* the colliding plugin's registered commands as an unintended side-effect.

So I would suggest either clearing the dict before raising the exception, or writing to a temporary dict and then committing it to a final dict later.

Reply:
If `RuntimeError` is raised, then there will be no instance so no problem. :-)

Reply:
Oh true, ha. This is a c'tor...